src/dataframe/dataframe.jl

```diff
@@ -337,6 +337,30 @@ function DataFrame(columns::AbstractMatrix, cnames::Symbol)
     return DataFrame(columns, gennames(size(columns, 2)), makeunique=false)
 end
 
+# Discontinued constructors
```

Comment:
I have reverted the constructors that "made sense" and were common-enough.

The left out tuple-constructors now are:

```
julia> DataFrame((:a => [1,2], :b => [3,4]))
2×2 DataFrame
 Row │ first   second 
     │ Symbol  Array… 
─────┼────────────────
   1 │ a       [1, 2]
   2 │ b       [3, 4]

julia> DataFrame(([1,2], [3,4]))
ERROR: ArgumentError: 'Tuple{Array{Int64,1},Array{Int64,1}}' iterates 'Array{Int64,1}' values, which doesn't satisfy the Tables.jl `AbstractRow` interface
```
but I do not think anyone used them.